pkg/streamingpromql/operators/subquery.go

```diff
@@ -115,9 +123,34 @@ func (s *Subquery) NextStepSamples() (*types.RangeVectorStepData, error) {
 	s.stepData.RangeStart = rangeStart
 	s.stepData.RangeEnd = rangeEnd
 
+	s.parentQueryStats.IncrementSamplesAtTimestamp(s.stepData.StepT, s.samplesProcessedInSubquery(s.stepData.RangeStart, s.stepData.RangeEnd))
+
 	return s.stepData, nil
 }
 
+// samplesProcessedInSubquery returns the number of samples processed in a time range selected by parent query step from subquery results.
+func (s *Subquery) samplesProcessedInSubquery(start, end int64) int64 {
+	if s.subqueryStats == nil || s.subqueryStats.TotalSamplesPerStep == nil || len(s.subqueryStats.TotalSamplesPerStep) == 0 {
+		return 0
+	}
+
+	startIndex := 0
+	if start >= s.SubqueryTimeRange.StartT {
```

Comment:
Why do we need this condition? (same applies for check on `endIndex >= len(s.subqueryStats.TotalSamplesPerStep)` below)

Reply:
For the startIndex check, it's a `dense_series{}[1m:5m]` case.
ParentQueryTimeRange is (600.000, 600.000). It gives a stepData Range (540.000, 600.000) 
But a SubQueryTimeRange is 600.000, 600.000. Which will give the incorrect start index when doing
`int((start-s.SubqueryTimeRange.StartT)/s.SubqueryTimeRange.IntervalMilliseconds) + 1`.

For the endIndex I don't have a case, but clamping it just to make sure we are always withing a stats array boundaries.

Reply:
> For the startIndex check, it's a `dense_series{}[1m:5m]` case.

Right, in that case, what if we did something like `start = max(start, s.SubqueryTimeRange.StartT)`? Then we could always use `PointIndex` below.

(It'd also be good to add a comment here either way so I don't ask the same question again in two months' time 🙂)

>For the endIndex I don't have a case, but clamping it just to make sure we are always withing a stats array boundaries.

If there's not a use case where we need to clamp it, I'd prefer not to clamp it, as that may hide a bug elsewhere.